Fabric.Identity.API/scripts/Install-Identity.ps1

```diff
@@ -0,0 +1,83 @@
+    $discoveryServiceUrl = Get-DiscoveryServiceUrl -discoveryServiceUrl $installSettings.discoveryService -quiet $quiet
+}
+$identityServiceUrl = Get-ApplicationEndpoint -appName $installSettings.appName -applicationEndpoint $installSettings.applicationEndPoint -quiet $quiet
+
+Unlock-ConfigurationSections
+$installApplication = Publish-Identity -site $selectedSite `
+                 -appName $installSettings.appName `
+                 -iisUser $iisUser `
+                 -zipPackage $zipPackage `
+
+Add-DatabaseSecurity $iisUser.UserName $installSettings.identityDatabaseRole $identityDatabase.DbConnectionString
+if(!$noDiscoveryService){
+    Register-IdentityWithDiscovery -iisUserName $iisUser.UserName -metadataConnStr $metadataDatabase.DbConnectionString -version $installApplication.version -identityServerUrl $identityServiceUrl
+}
+
+Set-IdentityEnvironmentVariables -appDirectory $installApplication.applicationDirectory `
+-primarySigningCertificateThumbprint $selectedCerts.SigningCertificate.Thumbprint `
+-encryptionCertificateThumbprint $selectedCerts.EncryptionCertificate.Thumbprint `
+-appInsightsInstrumentationKey $appInsightsKey `
+-applicationEndpoint $identityServiceUrl `
+-identityDbConnStr $identityDatabase.DbConnectionString`
+-discoveryServiceUrl $discoveryServiceUrl `
+-noDiscoveryService $noDiscoveryService
+
+$accessToken = ""
+
+if(Test-RegistrationComplete $identityServiceUrl) {
+    $accessToken = Get-AccessToken -authUrl $identityServiceUrl -clientId "fabric-installer" -scope "fabric/identity.manageresources" -secret $installSettings.fabricInstallerSecret
+}
+
+$registrationApiSecret = Add-RegistrationApiRegistration -identityServerUrl $identityServiceUrl -accessToken $accessToken
+$fabricInstallerSecret = Add-InstallerClientRegistration -identityServerUrl $identityServiceUrl -accessToken $accessToken -fabricInstallerSecret $installSettings.fabricInstallerSecret
+Add-SecureInstallationSetting "common" "fabricInstallerSecret" $fabricInstallerSecret $selectedCerts.SigningCertificate
+
+if (!$accessToken){
+    $accessToken = Get-AccessToken -authUrl $identityServiceUrl -clientId "fabric-installer" -scope "fabric/identity.manageresources" -secret $fabricInstallerSecret
+}
+
+$identityClientSecret = Add-IdentityClientRegistration -identityServerUrl $identityServiceUrl -accessToken $accessToken
+Add-SecureIdentityEnvironmentVariables -encryptionCert $selectedCerts.SigningCertificate `
+    -identityClientSecret $identityClientSecret `
+    -registrationApiSecret $registrationApiSecret `
+    -appDirectory $installApplication.applicationDirectory
+
+if ($fabricInstallerSecret){
+    Write-DosMessage -Level "Information" -Message "Please keep the following Fabric.Installer secret in a secure place, it will be needed in subsequent installations:"
+    Write-DosMessage -Level "Information" -Message "Fabric.Installer clientSecret: $fabricInstallerSecret"
+}
+
+if(!$quiet){
```

Comment:
If run in `quiet` mode, will this exit before the user has had a chance to write down secrets if they want to? If so, do we want that? I know the secrets are written to the appropriate app config files, but if something happens to go wrong there and the user needs them, what will be the repercussions? They will just need to re-run the script (potentially in non-quiet mode)?

Reply:
It will, but the user will have the option of redirecting output to a file if they choose. There is also another workaround, the user can reset the installer secret if needed.